packages/cdk-assets/bin/docker-credential-cdk-assets.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Docker Credential Helper to retrieve credentials based on an external configuration file.
+ * Supports loading credentials from ECR repositories and from Secrets Manager,
+ * optionally via an assumed role.
+ *
+ * The only operation currently supported by this credential helper at this time is the `get`
+ * command, which receives a domain name as input on stdin and returns a Username/Secret in
+ * JSON format on stdout.
+ *
+ * IMPORTANT - The credential helper must not output anything else besides the final credentials
+ * in any success case; doing so breaks docker's parsing of the output and causes the login to fail.
+ */
+
+import * as fs from 'fs';
+import { DefaultAwsClient } from '../lib';
+
+import { cdkCredentialsConfig, cdkCredentialsConfigFile, fetchDockerLoginCredentials } from '../lib/private/docker-credentials';
+
+async function main() {
+  // Expected invocation is [node, docker-credential-cdk-assets, get] with input fed via STDIN
+  // For other valid docker commands (store, list, erase), we no-op.
+  if (process.argv.length !== 3 || process.argv[2] !== 'get') {
+    process.exit(0);
+  }
+
+  const config = cdkCredentialsConfig();
+  if (!config) {
+    throw new Error(`unable to find CDK Docker credentials at: ${cdkCredentialsConfigFile()}`);
+  }
+
+  // Read the domain to fetch from stdin
+  let rawDomain = fs.readFileSync(0, { encoding: 'utf-8' }).trim();
+  rawDomain = rawDomain.includes('://') ? rawDomain : `https://${rawDomain}`;
+  const domain = new URL(rawDomain).hostname;
```

Comment:
Uh. Is this necessary? Looks like you're turning a hostname into a URL only to get the hostname out again ?

Could do with a comment motivating the extra hoola-hoop.

Reply:
The docker documentation is fairly unclear on the concrete specification for allowable inputs. _Typically_, I've seen this being invoked with the full URL, and the docs state an example input as `https://index.docker.io/v1`, which necessitates the URL parsing.

What I was trying to avoid was getting an input like `index.docker.io/v1`. It's not clear that's possible, but I'm also not sure it isn't. I _think_ perhaps the only item I got bare domains was my own input/testing, and I can just trust we can pass the input straight to `new URL`, but I was trying to be a bit defensive. I can remove, or leave a comment describing my paranoia. :)

Reply:
Describing seems good.